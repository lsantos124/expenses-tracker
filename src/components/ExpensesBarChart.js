// MyBarChart.js
import React from 'react';
import {
    FlexibleWidthXYPlot,
    XAxis, // Shows the values on x axis
    YAxis, // Shows the values on y axis
    VerticalBarSeries,
    LabelSeries,
    HorizontalGridLines
} from 'react-vis';

class ExpensesBarChart extends React.Component {
    render() {
    	let maxData = 0;

		this.props.data.map((data) => {
			maxData = Math.max(maxData, data['y']);
			return data;
		});
        
        const data = this.props.data;
        const chartHeight = 250;
        const chartDomain = [0, maxData + 20];
        return (
            <FlexibleWidthXYPlot 
                xType="ordinal" 
                height={chartHeight}
                yDomain={chartDomain}
            >
                <XAxis 
                	title="Date"
                	tickLabelAngle={45} 
                	tickPadding={35}
            	    tickValues={(
            	    	data.length > 15)
			              ? data
			                .filter((item, idx) => {
			                  if ((idx % Math.floor(data.length / 15)) === 0) {
			                    return item.x
			                  }
			                }).map(item => (item.x))
			              : data.map(item => (item.x))
			          }
			    />
                <YAxis title="Amount ($)"/>
                <VerticalBarSeries
                    data={data}
                />
                <LabelSeries
                    data={data.map(obj => {
                        return { ...obj, label: obj.y.toString() }
                    })}
                    labelAnchorX="start"
                    labelAnchorY="text-after-edge"
                    rotation={-45}
                />
                <HorizontalGridLines />
            </FlexibleWidthXYPlot>
        );
    }
}
export default ExpensesBarChart;