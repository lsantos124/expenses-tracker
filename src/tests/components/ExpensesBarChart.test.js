import React from 'react';
import { shallow } from 'enzyme';
import ExpensesBarChart from '../../components/ExpensesBarChart';

test('should render ExpensesBarChart with no expenses', () => {
	const wrapper = shallow(<ExpensesBarChart data={[]}/>);
	expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesBarChart with expenses', () => {
	const data = [
		{'y': 25, 'x':'12-31'},
		{'y': '', 'x':'01-01'},
		{'y': '', 'x':'01-02'},
		{'y': '', 'x':'01-03'},
		{'y': 16, 'x':'01-04'}
	];

	const wrapper = shallow(<ExpensesBarChart data={data}/>);
	expect(wrapper).toMatchSnapshot();
});