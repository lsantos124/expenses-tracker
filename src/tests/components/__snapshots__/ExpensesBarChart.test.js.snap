// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesBarChart with expenses 1`] = `
<FlexibleXYPlot
  height={250}
  xType="ordinal"
  yDomain={
    Array [
      0,
      45,
    ]
  }
>
  <XAxis
    attr="x"
    attrAxis="y"
    orientation="bottom"
    tickLabelAngle={45}
    tickPadding={35}
    tickValues={
      Array [
        "12-31",
        "01-01",
        "01-02",
        "01-03",
        "01-04",
      ]
    }
    title="Date"
  />
  <YAxis
    attr="y"
    attrAxis="x"
    orientation="left"
    title="Amount ($)"
  />
  <VerticalBarSeries
    className=""
    data={
      Array [
        Object {
          "x": "12-31",
          "y": 25,
        },
        Object {
          "x": "01-01",
          "y": "",
        },
        Object {
          "x": "01-02",
          "y": "",
        },
        Object {
          "x": "01-03",
          "y": "",
        },
        Object {
          "x": "01-04",
          "y": 16,
        },
      ]
    }
    stack={false}
    style={Object {}}
  />
  <LabelSeries
    animation={false}
    className=""
    data={
      Array [
        Object {
          "label": "25",
          "x": "12-31",
          "y": 25,
        },
        Object {
          "label": "",
          "x": "01-01",
          "y": "",
        },
        Object {
          "label": "",
          "x": "01-02",
          "y": "",
        },
        Object {
          "label": "",
          "x": "01-03",
          "y": "",
        },
        Object {
          "label": "16",
          "x": "01-04",
          "y": 16,
        },
      ]
    }
    getLabel={[Function]}
    labelAnchorX="middle"
    labelAnchorY="text-after-edge"
    rotation={0}
    stack={false}
    style={Object {}}
  />
  <HorizontalGridLines
    attr="y"
    direction="horizontal"
  />
</FlexibleXYPlot>
`;

exports[`should render ExpensesBarChart with no expenses 1`] = `
<FlexibleXYPlot
  height={250}
  xType="ordinal"
  yDomain={
    Array [
      0,
      20,
    ]
  }
>
  <XAxis
    attr="x"
    attrAxis="y"
    orientation="bottom"
    tickLabelAngle={45}
    tickPadding={35}
    tickValues={Array []}
    title="Date"
  />
  <YAxis
    attr="y"
    attrAxis="x"
    orientation="left"
    title="Amount ($)"
  />
  <VerticalBarSeries
    className=""
    data={Array []}
    stack={false}
    style={Object {}}
  />
  <LabelSeries
    animation={false}
    className=""
    data={Array []}
    getLabel={[Function]}
    labelAnchorX="middle"
    labelAnchorY="text-after-edge"
    rotation={0}
    stack={false}
    style={Object {}}
  />
  <HorizontalGridLines
    attr="y"
    direction="horizontal"
  />
</FlexibleXYPlot>
`;
